File: 2644. Find the Maximum Divisibility Score/solution.cpp
class Solution {
public:
  int maxDivScore(vector<int>& nums, vector<int>& divisors) {
    int divisibilityValue = divisors[0];
    int maxCount = 0;
    for (int index = 0; index < divisors.size(); index++){
      int divisor = divisors[index];
      int count = 0;
      for (int index2 = 0; index2 < nums.size(); index2++){
        int num = nums[index2];
        if (num % divisor == 0){
          count += 1;
        }
      }
      if (count > maxCount){
        divisibilityValue = divisor;
        maxCount = count;
      }
      else if (count == maxCount && divisor < divisibilityValue){
        divisibilityValue = divisor;
      }
    }
    return divisibilityValue;
  }
};

File: 2644. Find the Maximum Divisibility Score/solution.py
class Solution(object):
  def maxDivScore(self, nums, divisors):
    """
    :type nums: List[int]
    :type divisors: List[int]
    :rtype: int
    """
    divisibilityValue = divisors[0]
    maxCount = 0
    for divisor in divisors:
      count = 0
      for num in nums:
        if num % divisor == 0:
          count += 1
      if count > maxCount:
        divisibilityValue = divisor
        maxCount = count
      elif count == maxCount and divisor < divisibilityValue:
        divisibilityValue = divisor
    return divisibilityValue